Lib/test/test_ssl.py

```diff
@@ -1393,6 +1393,18 @@ def test_subclass(self):
                 # For compatibility
                 self.assertEqual(cm.exception.errno, ssl.SSL_ERROR_WANT_READ)
 
+    def test_bad_idna_in_server_hostname(self):
+        # Note: this test is testing some code that probably shouldn't exist
+        # in the first place, so if it starts failing at some point because
+        # you made the ssl module stop doing IDNA decoding then please feel
+        # free to remove it.
+        ctx = ssl.create_default_context()
+        # Do it a few times, because this used to cause memory corruption
+        # https://bugs.python.org/issue30594
+        for _ in range(5):
```

Comment:
I don't think that it's worth it to run it in a loop. Just run it once. It's only a regression test. We have enough buildbots to catch race conditions *if* there is a test ;-)